recipes/leptonica/all/conanfile.py

```diff
@@ -1,42 +1,68 @@
 import os
-import shutil
+import textwrap
+
+required_conan_version = ">=1.33.0"
 
 
 class LeptonicaConan(ConanFile):
     name = "leptonica"
     url = "https://github.com/conan-io/conan-center-index"
-    description = "Library containing software that is broadly useful for image processing and image analysis applications."
+    description = "Library containing software that is broadly useful for " \
+                  "image processing and image analysis applications."
     topics = ("conan", "leptonica", "image", "multimedia", "format", "graphics")
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package"
     homepage = "http://leptonica.org"
     license = "BSD 2-Clause"
-    exports_sources = ["CMakeLists.txt"]
-
-    options = {"shared": [True, False],
-               "with_gif": [True, False],
-               "with_jpeg": [True, False],
-               "with_png": [True, False],
-               "with_tiff": [True, False],
-               "with_openjpeg": [True, False],
-               "with_webp": [True, False],
-               "fPIC": [True, False]
-              }
-    default_options = {'shared': False,
-                       'with_gif': False,
-                       'with_jpeg': True,
-                       'with_png': True,
-                       'with_tiff': True,
-                       'with_openjpeg': True,
-                       'with_webp': True,
-                       'fPIC': True}
 
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False],
+        "with_gif": [True, False],
+        "with_jpeg": [True, False],
+        "with_png": [True, False],
+        "with_tiff": [True, False],
+        "with_openjpeg": [True, False],
+        "with_webp": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+        "with_gif": True,
+        "with_jpeg": True,
+        "with_png": True,
+        "with_tiff": True,
+        "with_openjpeg": True,
+        "with_webp": True
+    }
+
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package", "pkg_config"
     _cmake = None
-    _source_subfolder = "source_subfolder"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
 
     def requirements(self):
-        self.requires("zlib/1.2.11")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
```

Comment:
Comment about `self.requires("openjpeg/2.4.0")`:

Leptonica uses versioned paths (2.3, 2.4) in source files and cmake. To properly support openjpeg 2.4.0 we should replace `openjpeg-2.3` with `openjpeg-2.4`

Reply:
It doesn't use versioned paths if LIBJP2K_HEADER is not defined (which should be the case), so it should properly works.

Reply:
That's right, it should work.